PinScreen.js

```diff
@@ -190,9 +190,11 @@ class PinScreen extends Component {
       Vibration.vibrate(50);
     }
 
-    // The pin has been changed, trigger the callback
-    // Don't allow the callback if the input exceeds the number of pins
-    if (newPin.length <= numberOfPins) {
+    // Save the new pin into the state
+    this.setState({ pin: newPin });
+
+    // If the newPin matches the required length, perform the callback
+    if (newPin.length === numberOfPins) {
```

Comment:
I'm not sure that this change is the intended behaviour, could you explain more the about the reason behind it? The intended behaviour of `keyDown` is _"Callback function triggered when a key is pressed. Returns the key value."_, this change seems to change the behaviour to enable a callback when pin is complete. I suggest the following:

```suggestion
    // The pin has been changed, trigger the callback
    // Don't allow the callback if the input exceeds the number of pins
    if (newPin.length <= numberOfPins) {
         if (keyDown) keyDown(newPin);
    }

    // If the newPin matches the required length, perform the callback
    if (newPin.length === numberOfPins) {
          if(this.props.onComplete) this.props.onComplete();
    }
```

Maybe I am wrong here on what this fix addresses (as it's been a while since I worked on this lib), let me know what you think, would this solve your issues? It seems to me that you need a callback for when the pin screen is complete so you can perform an action.

Reply:
Yes, an onComplete would be sufficient in my case. I can't recall exactly why I changed `newPin.length <= numberOfPins)` to `newPin.length === numberOfPins)`,

`this.setState({ pin: newPin });` is used for `clearPin()`, so it's still required.

Reply:
@gerwim Are you using this as a controlled component?

```
const [pin, setPin] = useState("");

<PinScreen
    pin={pin}
    keyDown={setPin}
/>
```

Then you could just clear the pin by running `setPin("")` in your consuming code. Just trying to understand your use case.

Reply:
Yes, that's possible. However `pin` is not an exposed prop (atleast, not in the initial render) from what I see in PinScreen.

My project is not in active development anymore, so you can close this PR if you think the changes are not needed right now. I can reopen and recheck the code if my project goes in active state again and review both your code and my code, so I can understand my own use case aswell since it was some time ago ;-).

Reply:
Gotcha, I'm going to close this for now, as his change may break others code. If this issue comes back up, we can re-open the PR.